includes/Integrations/Jetpack.php

```diff
@@ -68,6 +95,153 @@ public function add_to_jetpack_sitemap( $post_types ) {
+			return $response;
+		}
+
+		// Reset mime type back to mp4, as this is the correct value.
+		$response['mime']    = 'video/mp4';
+		$response['subtype'] = 'mp4';
+		// Make video as optimized.
+		$response['media_source'] = 'video-optimization';
+
+		$media_details = wp_get_attachment_metadata( $attachment->ID );
+		if ( is_array( $media_details ) && isset( $media_details['videopress'] ) ) {
+			$videopress = $media_details['videopress'];
+			// If videopress has finished processing, use the duration in millions to get formatted seconds and minutes.
+			if ( isset( $videopress['duration'] ) && $videopress['duration'] ) {
+				$response['fileLength'] = $this->format_milliseconds( $videopress['duration'] );
+
+			}
+			// If video has not finished processing, reset request to original url.
+			if ( isset( $videopress['finished'], $videopress['original'] ) && ! $videopress['finished'] ) {
+				$response['url'] = $videopress['original'];
+			}
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Filter REST API responses to change video/videopress back to mp4.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     The original attachment post.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function filter_api_response( WP_REST_Response $response, WP_Post $post ) {
+		if ( self::VIDEOPRESS_MIME_TYPE !== $post->post_mime_type ) {
+			return $response;
+		}
+
+		$data = $response->get_data();
+
+		// Reset mime type back to mp4, as this is the correct value.
+		$data['mime_type'] = 'video/mp4';
+		// Make video as optimized.
+		$data['media_source'] = 'video-optimization';
+
+		if ( isset( $data['media_details']['videopress'] ) ) {
+			$videopress = $data['media_details']['videopress'];
+			// If videopress has finished processing, use the duration in millions to get formatted seconds and minutes.
+			if ( isset( $videopress['duration'] ) && $videopress['duration'] ) {
+				$data['media_details']['length_formatted'] = $this->format_milliseconds( $videopress['duration'] );
+			}
+			// If video has not finished processing, reset request to original url.
+			if ( isset( $videopress['finished'], $videopress['original'] ) && ! $videopress['finished'] ) {
+				$data['source_url'] = $videopress['original'];
+			}
+		}
+
+		$response->set_data( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Format milliseconds into seconds.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param int $milliseconds Milliseconds to converted to minutes and seconds.
+	 *
+	 * @return string
+	 */
+	protected function format_milliseconds( $milliseconds ) {
+		$seconds = floor( $milliseconds / 1000 );
+
+		if ( $seconds >= 1 ) {
+			$minutes  = floor( $seconds / 60 );
+			$seconds %= 60;
+		} else {
+			$seconds = 0;
+			$minutes = 0;
+		}
+
+		return sprintf( '%d:%02u', $minutes, $seconds );
+	}
+
+	/**
+	 * Hook into add term.
```

Comment:
```suggestion
	 * Hook into added_post_meta.
```

Maybe? A bit confusing otherwise.